File: boj/28702_FizzBuzz.js
const filePath = process.platform === 'linux' ? '/dev/stdin' : 'input.txt';
const input = require('fs').readFileSync(filePath).toString().trim().split('\n');

function solution(input) {
	const arr = [0, 0, 0];

	const index = input.findIndex((i) => !(i === 'Fizz' || i === 'Buzz' || i === 'FizzBuzz'));
	arr[index] = +input[index];
	for (let i = index + 1; i < input.length; i += 1) {
		arr[i] = arr[i - 1] + 1;
	}

	const last = arr[arr.length - 1] + 1;
	if (last % 3 === 0 && last % 5 === 0) return 'FizzBuzz';
	else if (last % 3 === 0 && last % 5 !== 0) return 'Fizz';
	else if (last % 3 !== 0 && last % 5 === 0) return 'Buzz';
	return last;
}

console.log(solution(input));

File: boj/28702_FizzBuzz.py
import sys

sys.stdin = open("./input.txt", 'r')
input_data = [l.strip() for l in sys.stdin.readlines()]


def solution(input_data):
    index = 0
    for idx, val in enumerate(input_data):
        if val != 'FizzBuzz' and val != 'Fizz' and val != 'Buzz':
            index = idx
    arr = [0, 0, 0]
    arr[index] = int(input_data[index])
    for i in range(index+1, 3):
        arr[i] = arr[i-1]+1

    last = arr[-1]+1
    if last % 3 == 0 and last % 5 == 0:
        return 'FizzBuzz'
    elif last % 3 == 0 and last % 5 != 0:
        return 'Fizz'
    elif last % 3 != 0 and last % 5 == 0:
        return 'Buzz'
    return last


print(solution(input_data))
